### PR TITLE
support consumption plan as a option

### PR DESCRIPTION
**Pull request by yiliuTo**
This is pr is to make the AZD template support both ASA consumption plan and standard plan, and the default option will be the consumption plan.

To test it, you can run:
```
mkdir gs-spring-boot-for-azure
cd gs-spring-boot-for-azure
azd auth login
azd config set alpha.springapp on
azd init --template https://github.com/yiliuTo/gs-spring-boot-for-azure/ -b support-consumption
azd up
```

The template by default uses consumption plan, and if users want to use standard, they can set `azd env set PLAN standard` then run `azd up`. If users have already provisioned the resources with the consumption plan and want to try the Standard plan, they need to run azd down first to delete the resources, and then run azd env set PLAN standard and azd up again to provision and deploy.

### CODE REVIEW

**Comment by yiliuTo** [comment]
We will not merge this pr until the corresponding MS doc gets merged. cc @moarychan and @Netyyyy

**Comment by moarychan** [comment]
@yiliuTo  MS doc PR has been merged, this PR is available to be merged now.

**Comment by yiliuTo** [comment]
Hi @fangjian0423 and @Buzzardo , could you help to review this pr? thanks

**Review by fangjian0423 — APPROVED** [review]
LGTM

**Comment by Buzzardo** [comment]
I'm not familiar enough with the code base to give a good review. I'll take Jimmy's word for it. Let me know when the document component is ready for review. Thanks.

**Pull request merged**
